src/compute/src/logging/compute.rs

```diff
@@ -133,387 +133,144 @@ pub fn construct<A: Allocate>(
-                                                "tried to create update frontier for import that doesn't exist"
-                                            );
-                                        }
-                                    }
-                                }
-                            }
-                            ComputeEvent::Peek(peek, is_install) => {
-                                let key = (worker, peek.uuid);
-                                if is_install {
-                                    peek_session.give(((peek, worker), time_ms, 1));
-                                    if peek_stash.contains_key(&key) {
-                                        error!(
-                                            "peek already registered: \
-                                             worker={}, uuid: {}",
-                                            worker, key.1,
-                                        );
-                                    }
-                                    peek_stash.insert(key, time.as_nanos());
-                                } else {
-                                    peek_session.give(((peek, worker), time_ms, -1));
-                                    if let Some(start) = peek_stash.remove(&key) {
-                                        let elapsed_ns = time.as_nanos() - start;
-                                        let elapsed_pow = elapsed_ns.next_power_of_two();
-                                        let elapsed_ns: i128 =
-                                            elapsed_ns.try_into().expect("elapsed_ns too big");
-                                        peek_duration_session.give((
-                                            (key.0, elapsed_pow),
-                                            time_ms,
-                                            (elapsed_ns, 1u64),
-                                        ));
-                                    } else {
-                                        error!(
-                                            "peek not yet registered: \
-                                             worker={}, uuid: {}",
-                                            worker, key.1,
-                                        );
-                                    }
-                                }
-                            }
+                    let mut output_sessions = DemuxOutput {
+                        export: export.session(&cap),
+                        dependency: dependency.session(&cap),
+                        frontier: frontier.session(&cap),
+                        import_frontier: import_frontier.session(&cap),
+                        frontier_delay: frontier_delay.session(&cap),
+                        peek: peek.session(&cap),
+                        peek_duration: peek_duration.session(&cap),
+                    };
+
+                    for (time, _logger_id, event) in demux_buffer.drain(..) {
```

Comment:
Although it's true, there is no guarantee that `_logger_id` equals the current worker index.